### PR TITLE
Implement clean-up items before 'Load' step

### PR DESCRIPTION
**Pull request by dladd**
Addresses #110

### CODE REVIEW

File: tests/coverage/coverage.cpp [hunk 1]
```diff
@@ -139,3 +148,35 @@ TEST(Coverage, variable) {
...
+    libcellml::Component p, pm;
+    libcellml::ComponentPtr child = std::make_shared<libcellml::Component>();
+    p.addComponent(child);
+    EXPECT_EQ(e, p.serialise(libcellml::Format::XML));
+
+    pm = std::move(p);
+    EXPECT_EQ(e, pm.serialise(libcellml::Format::XML));
+
+    libcellml::Component pc(pm);
+    EXPECT_EQ(e, pm.serialise(libcellml::Format::XML));
```
Comment (hsorby) [inline tests/coverage/coverage.cpp]: Probably need to test against the expected string of pc and not pm.

File: tests/coverage/coverage.cpp [hunk 2]
```diff
@@ -139,3 +148,35 @@ TEST(Coverage, variable) {
...
+    c.addVariable(v);
+    c.setMath("<1+1=2>");
+    EXPECT_EQ(e, c.serialise(libcellml::Format::XML));
+
+    cm = std::move(c);
+    EXPECT_EQ(e, cm.serialise(libcellml::Format::XML));
+
+    // Copy constructor
+    libcellml::Component cc(cm);
+    EXPECT_EQ(e, cm.serialise(libcellml::Format::XML));
```
Comment (hsorby) [inline tests/coverage/coverage.cpp]: Probably need to test against the expected string of cc and not cm.

**Comment by agarny** [comment]
Besides @hsorby's comments, it all looks good to me.

**Comment by dladd** [comment]
Thanks all- this now addresses comments, passes tests, and fulfils items discussed in #110

**Pull request merged**